src/net.rs

```diff
@@ -230,7 +270,10 @@ pub fn test_read_commands(){
     // read the command from the stream for Command::Quit
     let mut command_res = read_commands(&mut Cursor::new(vec));
     assert_eq!(command_res.is_ok(), true);
-    assert_eq!(command_res.unwrap(), Command::Quit);
+    match command_res.unwrap() {
+        Command::Quit => assert!(true),
+        _ => assert!(false)
+    };
```

Comment:
Instead of the match block, maybe:
```
assert_eq!(command_res.unwrap(), Command::Quit);
```
Maybe `Command` need to implement `PartialEq`